File: 백준/Silver/1541. 잃어버린 괄호/잃어버린 괄호.py
import sys


exp = input().split("-")
num = []

for i in exp:
    total = 0
    total += sum(map(int, i.split("+")))
    num.append(total)

result = num[0]
for i in range(1, len(num)):
    result -= num[i]

print(result)

File: 백준/Silver/1541. 잃어버린 괄호/잃어버린 괄호.java

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main {

	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		int result = Integer.MAX_VALUE;
		String exp = br.readLine();
		String[] spl = exp.split("-");
		for (int i = 0; i < spl.length; i++) {
			int tmp = 0;
			String[] sumSpl = spl[i].split("\\+");
			for (int j = 0; j < sumSpl.length; j++) {
				tmp += Integer.parseInt(sumSpl[j]);
			}

			if (result == Integer.MAX_VALUE) {
				result = tmp;
			} else {
				result -= tmp;
			}
		}
		System.out.println(result);
	}

}